processing/src/main/java/io/druid/query/groupby/GroupByQuery.java

```diff
@@ -136,14 +136,14 @@ public GroupByQuery(
         this.limitSpec.build(this.dimensions, this.aggregatorSpecs, this.postAggregatorSpecs);
 
     if (havingSpec != null) {
-      havingSpec.setRowType(GroupByQueryHelper.rowTypeFor(this));
       postProcFn = Functions.compose(
           postProcFn,
           new Function<Sequence<Row>, Sequence<Row>>()
           {
             @Override
             public Sequence<Row> apply(Sequence<Row> input)
             {
+              GroupByQuery.this.havingSpec.setRowSignature(GroupByQueryHelper.rowSignatureFor(GroupByQuery.this));
```

Comment:
I think this is a good idea although not needed for this PR. It would be a good change if we ever _do_ need to use HavingSpecs in multiple different threads.